### ISSUE TITLE
Extraneous trailing space when breaking long type definitions

### ISSUE THREAD

**Issue by rivy** [dprint/dprint-plugin-typescript#124]
**Describe the bug**

dprint-plugin-typescript version: 0.12.0

**Input Code**

```ts
type EnumerableValueOfT<T> = T extends [unknown, infer V][] ? V
	: T extends
		| ArrayLike<infer V>
		| Iterable<infer V>
		| AsyncIterable<infer V>
		| IterableIterator<infer V>
		| AsyncIterableIterator<infer V> ? V
	: unknown;
```

**Expected Output**

```ts
type EnumerableValueOfT<T> = T extends [unknown, infer V][] ? V
	: T extends
		| ArrayLike<infer V>
		| Iterable<infer V>
		| AsyncIterable<infer V>
		| IterableIterator<infer V>
		| AsyncIterableIterator<infer V> ? V
	: unknown;
```

**Actual Output**

```ts
type EnumerableValueOfT<T> = T extends [unknown, infer V][] ? V
	: T extends␠ 
		| ArrayLike<infer V>
		| Iterable<infer V>
		| AsyncIterable<infer V>
		| IterableIterator<infer V>
		| AsyncIterableIterator<infer V> ? V
	: unknown;
```

The `: T extends` line maintains an incorrect trailing space.

**Comment by dsherret** [comment]
Thanks for opening! I investigated this, opened & resolved https://github.com/dprint/dprint/issues/350 as a fix, and I will do a new dprint-core release soon, which when used will fix this. Just want to get some other changes in before doing another release. For now, you can explicitly ignore this via a `// dprint-ignore` comment.

**Comment by rivy** [comment]
I compiled `dprint` from a2ff7af using `cargo +stable-x86_64` within **`crates/dprint`**. When formatting with the new executable, it still exhibits the same behavior, leaving a trailing space as before.

**Comment by dsherret** [comment]
@rivy the fix is in the `formatting` feature of dprint-core (crates/core in the dprint repo) which is compiled into the .wasm  file for this plugin. The CLI (crates/dprint) has no knowledge of the formatting feature of that crate.

If you really wanted to, you could point dprint-plugin-typescript at that commit of crates/core and then compile dprint-plugin-typescript with `cargo build --target wasm32-unknown-unknown --features "wasm" --release` to create a .wasm file which you can specify via a file path in the “plugins” array in your dprint.json file. Then this could be run with the latest released version of the dprint CLI.

Don’t worry though. I will do a release over the weekend.

**Comment by rivy** [comment]
> @rivy the fix is in the `formatting` feature of dprint-core (crates/core in the dprint repo) which is compiled into the .wasm file for this plugin. The CLI (crates/dprint) has no knowledge of the formatting feature of that crate.

Ah, thanks.
 
> Don’t worry though. I will do a release over the weekend.

No worry or rush; I was just going to try it out.

Thanks, again.

**Comment by rivy** [comment]
@dsherret , unfortunately, the trailing space is still added in `dprint 0.13.0`.

```ts
type EnumerableKeyOfT<T> = T extends [infer K, unknown][] ? K
	: T extends ArrayLike<unknown> ? number
	: T extends Set<infer K> ? K
	: T extends MapLikeObject<infer K, unknown> | MapLike<infer K, unknown> ? K
	: T extends␠
		| AnyGenerator<unknown, unknown, unknown>
		| AnyIterable<unknown>
		| AnyIterator<unknown> ? number
	: T extends Enumerable<unknown, infer K, unknown> ? K
	: unknown;
```

with `.dprint.json`:

```json
{
	"$schema": "https://dprint.dev/schemas/v0.json",
	"projectType": "openSource",
	"incremental": true,
	"indentWidth": 2,
	"lineWidth": 100,
	"useTabs": true,
	"typescript": {
		// ref: <https://dprint.dev/plugins/typescript/config>
		// * custom preferences
		"quoteStyle": "preferSingle",
		// * mirror *much of* Deno configuration
		// ref: <https://github.com/dprint/dprint-plugin-typescript/blob/44b6cf562e511a308f4a7183dc98fb19cdf21d07/src/configuration/builder.rs#L51>
		// ref: <https://github.com/denoland/deno/blob/f46e39c5c5/cli/tools/fmt.rs#L311>
		"nextControlFlowPosition": "sameLine",
		"binaryExpression.operatorPosition": "sameLine",
		"bracePosition": "sameLine",
		"commentLine.forceSpaceAfterSlashes": false,
		"constructSignature.spaceAfterNewKeyword": true,
		"constructorType.spaceAfterNewKeyword": true,
		"arrowFunction.useParentheses": "force",
		"newLineKind": "lf",
		"functionExpression.spaceAfterFunctionKeyword": true,
		"taggedTemplate.spaceBeforeLiteral": false
		// "conditionalExpression.preferSingleLine": true
		// "quoteStyle": "preferDouble",
		// "ignoreNodeCommentText": "deno-fmt-ignore",
		// "ignoreFileCommentText": "deno-fmt-ignore-file",
		// "module.sortImportDeclarations": "maintain",
		// "module.sortExportDeclarations": "maintain"
	},
	"json": {},
	"markdown": {},
	"rustfmt": {},
	"includes": ["**/*.{ts,tsx,js,jsx,json,md,rs}"],
	"excludes": [".history", "**/node_modules", "**/*-lock.json", "**/target", "**/vendor"],
	"plugins": [
		"https://plugins.dprint.dev/typescript-0.40.3.wasm",
		"https://plugins.dprint.dev/json-0.8.0.wasm",
		"https://plugins.dprint.dev/markdown-0.5.1.wasm",
		"https://plugins.dprint.dev/rustfmt-0.3.0.wasm"
	]
}
```

**Comment by dsherret** [comment]
@rivy the CLI releases are separate from the plugins. I haven't done a release for this plugin yet: https://github.com/dprint/dprint-plugin-typescript/releases (I'm planning on doing it later today)

That config file is showing it's on 0.40.3 of the dprint-plugin-typescript plugin (`"https://plugins.dprint.dev/typescript-0.40.3.wasm"`)

**Comment by dsherret** [comment]
By the way, if you want to mirror the deno configuration, you can start from the following in the configuration file...

```ts
"typescript": {
    "deno": true
}
```

...then add more configuration changes as you would like.

**Comment by dsherret** [comment]
This is now released in 0.44.1: https://github.com/dprint/dprint-plugin-typescript/releases/tag/0.44.1

**Comment by rivy** [comment]
Thanks!
